docs/docs/tutorials/codealong/simple_dapp/3_contract_interaction.md

```diff
@@ -18,53 +18,57 @@ To do this, let's first initialize a new `Contract` instance using `aztec.js` th
 ```
 
-And then add the following code for initializing the `Contract` instances:
+You may have noticed that we are importing the `TokenContract` class from `@aztec/noir-contracts.js`. This is an alternative way to get the contract interface for interacting with the contract. With this, we can add the following code for initializing the `TokenContract` instance:
 
 #include_code get-tokens yarn-project/end-to-end/src/sample-dapp/contracts.mjs javascript
 
-:::info
-You can use the typescript autogenerated interface instead of the generic `Contract` class to get type-safe methods.
-:::
+We can now get the token instance in our main code in `src/index.mjs`, by importing the function from `src/contracts.mjs`. Update the imports in `src/index.mjs` to look like this:
+
+```js
+// src/index.mjs
+#include_code imports yarn-project/end-to-end/src/sample-dapp/index.mjs javascript
+```
 
-We can now get the token instance in our main code in `src/index.mjs`, and query the private balance for each of the user accounts. To query a function, without sending a transaction, use the `view` function of the method:
+and query the private balance for each of the user accounts. To query a function, without sending a transaction, use the `simulate` function of the method:
 
 #include_code showPrivateBalances yarn-project/end-to-end/src/sample-dapp/index.mjs javascript
 
-Run this as `node src/index.mjs` and you should now see the following output:
+Call the function in `main` and run this with `node src/index.mjs` and you should now see the following output:
 
 ```
-Balance of 0x0c8a6673d7676cc80aaebe7fa7504cf51daa90ba906861bfad70a58a98bf5a7d: 100
+Balance of 0x0c8a6673d7676cc80aaebe7fa7504cf51daa90ba906861bfad70a58a98bf5a7d: 0
 Balance of 0x226f8087792beff8d5009eb94e65d2a4a505b70baf4a9f28d33c8d620b0ba972: 0
 Balance of 0x0e1f60e8566e2c6d32378bdcadb7c63696e853281be798c107266b8c3a88ea9b: 0
 ```
 
-## Transferring private tokens
+## Mint tokens
 
-Now that we can see the balance for each user, let's transfer tokens from one account to another. To do this, we will first need access to a `Wallet` object. This wraps access to an PXE and also provides an interface to craft and sign transactions on behalf of one of the user accounts.
+Before we can transfer tokens, we need to mint some tokens to our user accounts. Add the following function to `src/index.mjs`:
 
-We can initialize a wallet using one of the `getAccount` methods from the `accounts` package, along with the corresponding signing and encryption keys:
+#include_code mintPrivateFunds yarn-project/end-to-end/src/sample-dapp/index.mjs javascript
 
-```js
-import { getSchnorrAccount } from "@aztec/accounts/schnorr";
-const wallet = await getSchnorrAccount(
-  client,
-  ENCRYPTION_PRIVATE_KEY,
-  SIGNING_PRIVATE_KEY
-).getWallet();
+Call the function in `main`, run the script and you should see 20 tokens in the balance of the first account.
```

Comment:
```suggestion
Call the function in `main`, run the script and after printing the balances of each account it will then privately mint tokens. After that completes, you should then see 20 tokens in the balance of the first account.
```